src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -185,6 +189,34 @@ private function extractFromAccessor(string $class, string $property): ?array
         return null;
     }
 
+    /**
+     * Tries to extract type information from constructor.
+     *
+     * @return Type[]|null
+     */
+    private function extractFromConstructor(string $class, string $property): ?array
+    {
+        try {
+            $constructor = (new \ReflectionClass($class))->getConstructor();
+        } catch (\ReflectionException $e) {
+            return null;
+        }
+
+        if (!$constructor) {
```

Comment:
Should we take parent classes into account too?

Reply:
@lyrixx WDYT? How does phpstorm handle overridden constructors? It could help decide.

Reply:
I think we should take parents into account.
But I don't know how phpstorm handle that. I don't use it. You should know it ;)
I will check

Reply:
Just checked, PHP storm is using parent constructor too. I will update the PR

Reply:
Actually, it's already supported ;) I added a new test case to ensure it works as expected.

Reply:
I was thinking about this situation:
```php
class a
{
  public $foo;
  function __construct(string $foo)
  {
    $this->foo = $foo;
  }
}

class b extends a
{
  function __construct()
  {
    parent::__construct('hello');
  }
}
```

Reply:
Ah ok. I have added support for such case with a test case.

I guess the PR is now READY \o/